Should GetSelection panic or return nil?

`GetSelection` function returns `*Bucket`, but in case of internal errors it can panic. For example if there is a `Select` rule bigger than available `*Bucket`

```go
func TestBucket_RuntimeError(t *testing.T) {
	g := NewGomegaWithT(t)
	buckets := []bucket{
		{"/Location:Europe/Country:Spain/City:Madrid", []uint32{17, 18}},
	}
	root, err := newRoot(buckets...)
	g.Expect(err).NotTo(HaveOccurred())

	ss := []Select{
		{Key: NodesBucket, Count: 3}, // available only two
	}

	r := root.GetSelection(ss, defaultPivot)
	g.Expect(r).NotTo(BeNil())
}
```

This code leads to panic
```
=== RUN   TestBucket_RuntimeError
--- FAIL: TestBucket_RuntimeError (0.00s)
panic: runtime error: slice bounds out of range [recovered]
	panic: runtime error: slice bounds out of range
```

In our code we avoid this case by calling `GetMaxSelection` in advance. This function returns error if something goes wrong. But in general I don't know whether is it correct behaviour for `GetSelection` function to panic.

--- a/policy.go
+++ b/policy.go
@@ -277,6 +277,10 @@ func (b Bucket) GetSelection(ss []Select, pivot []byte) *Bucket {
 
 	count = int(ss[0].Count)
 	if ss[0].Key == NodesBucket {
+		if len(b.nodes) < count {
+			return nil
+		}
+
 		root.nodes = make(Uint32Slice, len(b.nodes))
 		copy(root.nodes, b.nodes)
 		if len(pivot) != 0 {

--- a/policy_test.go
+++ b/policy_test.go
@@ -26,6 +26,22 @@ func newRoot(bs ...bucket) (b Bucket, err error) {
 	return
 }
 
+func TestBucket_RuntimeError(t *testing.T) {
+	g := NewGomegaWithT(t)
+	buckets := []bucket{
+		{"/Location:Europe/Country:Spain/City:Madrid", []uint32{17, 18}},
+	}
+	root, err := newRoot(buckets...)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	ss := []Select{
+		{Key: NodesBucket, Count: 3}, // available only two
+	}
+
+	r := root.GetSelection(ss, defaultPivot)
+	g.Expect(r).To(BeNil())
+}
+
 func TestBucket_IsValid(t *testing.T) {
 	var (
 		b       Bucket

--- a/policy_test.go
+++ b/policy_test.go
@@ -58,6 +58,7 @@ func TestBucket_RuntimeError(t *testing.T) {
 		}
 
 		r := root.GetSelection(ss, defaultPivot)
+		g.Expect(r).NotTo(BeNil())
 		g.Expect(r.nodes).To(HaveLen(3))
 	})
 }

--- a/policy_test.go
+++ b/policy_test.go
@@ -27,19 +27,39 @@ func newRoot(bs ...bucket) (b Bucket, err error) {
 }
 
 func TestBucket_RuntimeError(t *testing.T) {
-	g := NewGomegaWithT(t)
-	buckets := []bucket{
-		{"/Location:Europe/Country:Spain/City:Madrid", []uint32{17, 18}},
-	}
-	root, err := newRoot(buckets...)
-	g.Expect(err).NotTo(HaveOccurred())
+	t.Run("slice bounds out of range", func(t *testing.T) {
+		g := NewGomegaWithT(t)
+		buckets := []bucket{
+			{"/Location:Europe/Country:Spain/City:Madrid", []uint32{17, 18}},
+		}
+		root, err := newRoot(buckets...)
+		g.Expect(err).NotTo(HaveOccurred())
 
-	ss := []Select{
-		{Key: NodesBucket, Count: 3}, // available only two
-	}
+		ss := []Select{
+			{Key: NodesBucket, Count: 3}, // available only two
+		}
 
-	r := root.GetSelection(ss, defaultPivot)
-	g.Expect(r).To(BeNil())
+		r := root.GetSelection(ss, defaultPivot)
+		g.Expect(r).To(BeNil())
+	})
+
+	t.Run("regression test", func(t *testing.T) {
+		g := NewGomegaWithT(t)
+		buckets := []bucket{
+			{"/Location:Europe/Country:Spain/City:Madrid", []uint32{17, 18}},
+			{"/Location:Europe/Country:Spain/City:Barcelona", []uint32{16, 19}},
+		}
+		root, err := newRoot(buckets...)
+		g.Expect(err).NotTo(HaveOccurred())
+
+		ss := []Select{
+			{Key: "Location", Count: 1},
+			{Key: NodesBucket, Count: 3},
+		}
+
+		r := root.GetSelection(ss, defaultPivot)
+		g.Expect(r.nodes).To(HaveLen(3))
+	})
 }
 
 func TestBucket_IsValid(t *testing.T) {

